kclvm/tools/src/LSP/src/util.rs

```diff
@@ -651,3 +651,45 @@ pub(crate) fn get_real_path_from_external(
     pkgpath.split('.').for_each(|s| real_path.push(s));
     real_path
 }
+
+/// Error recovery may generate an Identifier with an empty string at the end, e.g.,
+/// a. => vec["a", ""].
+/// When analyzing in LSP,the empty string needs to be removed and find definition of the second last name("a").
+pub(crate) fn fix_missing_identifier(names: &[String]) -> Vec<String> {
+    if names.len() >= 1 && names.last().unwrap() == "" {
+        return names[..names.len() - 1].to_vec();
+    } else {
+        return (names).to_vec();
+    }
+}
+
+pub(crate) fn pre_process_identifier(id: Node<Identifier>, pos: &KCLPos) -> Identifier {
+    if !id.contains_pos(pos) {
```

Comment:
Why the identifier process meets the `pos` para?

Reply:
After obtaining a complete identifer based on the location information, pos may be one of multiple names. The identifer needs to be modified according to pos. For example, pos is located at b in identifier `a.b.c`, we need to modify the identifier to `a.b`.